### PR TITLE
Cps/migrating check wake condition function

### PR DESCRIPTION
**Pull request by inigolcanalejo**
In this PR the check for the wake condition is moved from the element to the wake process.

### CODE REVIEW

**Comment by philbucher** [comment]
this seems quite small, why not making it a fct in the utilities?

File: applications/CompressiblePotentialFlowApplication/custom_processes/define_2d_wake_process.cpp [hunk 1]
```diff
@@ -318,4 +323,25 @@ const BoundedVector<double, 3> Define2DWakeProcess::ComputeDistanceFromTrailingE
     return distance_to_point;
 }
 
+// This function checks that the wake condition is fulfilled up to a certain tolerance
+void Define2DWakeProcess::ExecuteFinalizeSolutionStep()
```
Comment (marcnunezc) [inline applications/CompressiblePotentialFlowApplication/custom_processes/define_2d_wake_process.cpp]: Do you need this one defined here as well? Can you call the process directly?

**Review by marcnunezc — COMMENTED** [review]
I like the idea of moving the check from the element, but I agree with @philbucher, I would add it as a utility. Any reason no to do so that we did not think of?

File: applications/CompressiblePotentialFlowApplication/custom_processes/define_2d_wake_process.cpp (hunk 1 above)
Comment (inigolcanalejo) [reply applications/CompressiblePotentialFlowApplication/custom_processes/define_2d_wake_process.cpp]: Yes you are right. I will move the process to a utility and call it directly from here.

**Comment by inigolcanalejo** [comment]
Hi guys thanks for the review. I actually made a process because I wanted to expose it to python but actually I can do the same for a utility. I will add it to the utilities and let you know when I am done with the changes.

**Comment by philbucher** [comment]
I still think a utility is overkill, I suggest to use a free function

**Comment by inigolcanalejo** [comment]
@philbucher and @marcnunezc I think this is what you meant. Please let me know if you have any further suggestions.

@marcnunezc with this utility we can check how well ( well = with what tolerance) the wake condition is fulfilled. I just did a check in 2D and for a big naca0012 case the condition is fulfilled in all of the elements with an absolute tolerance of 1e-8 and for a small 2D case like the one in the test with a tolerance of 1e-13. For 3D however the condition is not fulfilled in some elements even with an absolute tolerance of 1e-1. Still the pressure distribution and the value of the lift coefficient looks ok for the cases I have run for infinite wings. But it is something we might want to bear in mind or further investigate.

File: applications/CompressiblePotentialFlowApplication/custom_utilities/potential_flow_utilities.cpp [hunk 2]
```diff
@@ -11,6 +11,9 @@
 
 #include "custom_utilities/potential_flow_utilities.h"
 #include "compressible_potential_flow_application_variables.h"
+#include "includes/model_part.h"
+#include "custom_elements/incompressible_potential_flow_element.h"
+#include "compressible_potential_flow_application_variables.h"
```
Comment (philbucher) [inline applications/CompressiblePotentialFlowApplication/custom_utilities/potential_flow_utilities.cpp]: this is duplicated

File: applications/CompressiblePotentialFlowApplication/custom_utilities/potential_flow_utilities.cpp [hunk 3]
```diff
@@ -11,6 +11,9 @@
 
 #include "custom_utilities/potential_flow_utilities.h"
 #include "compressible_potential_flow_application_variables.h"
+#include "includes/model_part.h"
+#include "custom_elements/incompressible_potential_flow_element.h"
```
Comment (philbucher) [inline applications/CompressiblePotentialFlowApplication/custom_utilities/potential_flow_utilities.cpp]: why do you need this include?

**Review by philbucher — COMMENTED** [review]
looks good to me! (minor comment)

File: applications/CompressiblePotentialFlowApplication/custom_utilities/potential_flow_utilities.cpp (hunk 3 above)
Comment (inigolcanalejo) [reply applications/CompressiblePotentialFlowApplication/custom_utilities/potential_flow_utilities.cpp]: I don't ;-)

**Review by marcnunezc — APPROVED** [review]
Ok to me!

What I would do eventually is adding some way to enable/disable the check to avoid the overhead if desired, but that can be a part of a new PR.

**Comment by inigolcanalejo** [comment]
> Ok to me!
> 
> What I would do eventually is adding some way to enable/disable the check to avoid the overhead if desired, but that can be a part of a new PR.

I agree, I think we can run this utility from the solver and use the echo level to decide whether to run it or not.

Also the check will most probably need to be adapted when we implement the condition in 3d to check only the components of the velocity in the free stream and wake normal direction.

**Pull request merged**